homeassistant/components/lektrico/coordinator.py

```diff
@@ -0,0 +1,52 @@
+"""Coordinator for the Lektrico Charging Station integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+
+from aiohttp import ClientSession
+from lektricowifi import DeviceConnectionError, lektricowifi
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import LOGGER
+
+SCAN_INTERVAL = timedelta(seconds=10)
+
+
+class LektricoDeviceDataUpdateCoordinator(DataUpdateCoordinator):
+    """The device class for Lektrico device."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        friendly_name: str,
+        host: str,
+        session: ClientSession,
+        serial_number: str,
+        board_revision: str,
+        device_type: str,
+    ) -> None:
+        """Initialize a Lektrico Device."""
+        self.device = lektricowifi.Device(
+            host,
+            session=session,
+        )
+        self.friendly_name = friendly_name.replace(" ", "_")
+        self._name = friendly_name
```

Comment:
This is a) never used and b) confusing because `DataUpdateCoordinator` already has a name attribute